src/App.js

```diff
@@ -1,25 +1,68 @@
-import logo from './logo.svg';
-import './App.css';
+// import PropTypes from 'prop-types'
+import styled from 'styled-components/macro'
+import { useState } from 'react'
+import Grid from './components/Grid'
+import AddPlayerForm from './components/AddPlayerForm/AddPlayerForm'
+import Player from './components/Player'
+import Button from './components/Button'
+
+export default function App() {
+  const [players, setPlayers] = useState([])
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    <Grid>
+      <AddPlayerForm
+        role="form"
+        name="addPlayerForm"
+        onAddPlayer={addPlayer}
+        disabled={players.length >= 4 ? true : false}
+        placeholderText={
+          players.length >= 4
+            ? 'Your flight has reached the max. of 4 players'
+            : 'Players name goes here (min. 2 players)'
+        }
+      />
+      {players.map(({ name, score }, index) => (
+        <Player
+          key={index}
+          players={name}
+          score={score}
+          onScore={() => countScore(index)}
+          disabled={players.length <= 1 ? true : false}
+        />
+      ))}
+      <ResetButton
+        disabled={players.length <= 0 ? true : false}
+        onClick={resetForm}
+      >
+        Reset
+      </ResetButton>
+    </Grid>
+  )
+
+  function addPlayer({ nameOfPlayer }) {
+    setPlayers([{ name: nameOfPlayer, score: 0 }, ...players])
+  }
+
+  function countScore(index) {
+    const currentPlayer = players[index]
+    setPlayers([
+      ...players.slice(0, index),
+      { ...currentPlayer, score: currentPlayer.score + 1 },
+      ...players.slice(index + 1),
+    ])
+  }
+
+  function resetForm() {
+    setPlayers([])
+  }
 }
 
-export default App;
+export const ResetButton = styled(Button)`
```

Comment:
Are you planning on exporting the ResetButton for another component? Maybe you wouldn't need a separate styled component if you use a button wrapper with `display: flex` or `grid` instead of absolute positioning? But that design choice is yours entirely, of course! ☺️